sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,340 @@
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
+            raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+
+        def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
+            result_data = []
+            result_index = []
+
+            for dataframe_item in self._data:
+                item_count = dataframe_item.count()
+                item_name = dataframe_item._name
+                result_data.append(item_count)
+                result_index.append(item_name)
+
+            return pandas.Series(data=result_data, index=result_index)
+
+        return sdc_pandas_dataframe_count_impl
+
+else:
+    def reduce(df, name):
+        saved_columns = df.columns
+        n_cols = len(saved_columns)
+        data_args = tuple('data{}'.format(i) for i in range(n_cols))
+        func_text = "def _reduce_impl(df, axis=None, skipna=None, level=None, numeric_only=None):\n"
+        for i, d in enumerate(data_args):
+            func_text += "  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))\n".format(
+                d + '_S', i)
+            func_text += "  {} = {}.{}()\n".format(d + '_O', d + '_S', name)
+        func_text += "  data = np.array(({},))\n".format(
+            ", ".join(d + '_O' for d in data_args))
+        func_text += "  index = hpat.str_arr_ext.StringArray(({},))\n".format(
+            ", ".join("'{}'".format(c) for c in saved_columns))
+        func_text += "  return hpat.hiframes.api.init_series(data, index)\n"
+        loc_vars = {}
+
+        print()
+        print(func_text)
+        print()
+
+        exec(func_text, {'hpat': sdc, 'np': numpy}, loc_vars)
+        _reduce_impl = loc_vars['_reduce_impl']
+
+        return _reduce_impl
+
+
+    @overload_method(DataFrameType, 'median')
+    def median_overload(df, axis=None, skipna=None, level=None, numeric_only=None):
+        """
+           Pandas DataFrame method :meth:`pandas.DataFrame.median` implementation.
+
+           .. only:: developer
+
+               Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_median1
+
+           Parameters
+           -----------
+           self: :class:`pandas.DataFrame`
+               input arg
+           axis:
+                *unsupported*
+           skipna:
+               *unsupported*
+           level:
+               *unsupported*
+           numeric_only:
+               *unsupported*
+
+           Returns
+           -------
+           :obj:`pandas.Series` or `pandas.DataFrame`
+                   return the median of the values for the requested axis.
+           """
+
+        name = 'median'
+
+        if not isinstance(df, DataFrameType):
+            raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(name, df))
+
+        if not (isinstance(axis, types.Omitted) or axis is None):
```

Comment:
Probably we should use ty_checker here and everywhere else, e.g.
```suggestion
    if (not isinstance(axis, (int, types.Integer, str, types.UnicodeType, types.StringLiteral, types.Omitted))
        and axis not in (0, 'index')):
        ty_checker.raise_exc(axis, 'integer or string', 'axis')
```

Reply:
All functions seems to have more or less the same checks. Shouldn't we move generic checks into generic function in order to avoid copy and paste?